IOS_SYMBOLICATION.md

```diff
@@ -0,0 +1,76 @@
+# Symbolicating iOS crash reports
+
+Debugging an iOS application crash sometimes involves analyzing crash reports.
+More info about crash reports can be found
+[in the official documentation](https://developer.apple.com/library/archive/technotes/tn2151/_index.html).
+
+Crash reports generally require symbolication to become properly readable:
+symbolication turns raw hex addresses into human-readable source locations.
+The document below describes some specific details of symbolicating crash reports
+from iOS applications using Kotlin.
+
+## Enable .dSYM for release Kotlin binaries
+
+To symbolicate addresses in Kotlin code (e.g. for stack trace elements
+corresponding to Kotlin code) `.dSYM` bundle for Kotlin code is required.
+
+By default Kotlin/Native compiler doesn't produce `.dSYM` for release
+(i.e. optimized) binaries. This can be changed with `-Xg0` experimental
+compiler flag: it enables debug info and `.dSYM` bundle generation for produced
+release binaries. To enable it in Gradle, use
+
+```kotlin
+this.targets.forEach {
+    if (it is KotlinNativeTarget) {
+        it.compilations.all {
+            kotlinOptions {
+                freeCompilerArgs += "-Xg0"
+            }
+        }
+    }
+}
+```
+
+(in Kotlin DSL).
+
+In projects created from IntelliJ IDEA or AppCode templates these `.dSYM` bundles
+are then discovered by Xcode automatically.
+
+## Make frameworks static when using rebuild from bitcode
+
+Rebuilding Kotlin-produced framework from bitcode invalidates the original `.dSYM`.
+If it is performed locally, make sure the updated `.dSYM` is used when symbolicating
+crash reports.
+
+If rebuilding is performed on App Store side, then `.dSYM` of rebuilt *dynamic* framework
+seems discarded and not downloadable from App Store Connect.
```

Comment:
and *is* not

Reply:
I don't think so.
"Seems not downloadable" is exactly what I mean here.